### PR TITLE
Replace config.sub and config.guess

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: conda_forge_tick/migrators/cross_compile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from conda_forge_tick.recipe_parser import CondaMetaYAML
+from conda_forge_tick.migrators.core import (
+    MiniMigrator,
+    _get_source_code,
+)
+
+LOGGER = logging.getLogger("conda_forge_tick.migrators.cross_compile")
+
+
+class LicenseMigrator(MiniMigrator):
```
Comment (beckermr) [inline conda_forge_tick/migrators/cross_compile.py:22]: name is odd :)

File: conda_forge_tick/migrators/cross_compile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+LOGGER = logging.getLogger("conda_forge_tick.migrators.cross_compile")
+
+
+class UpdateConfigSubGuessMigrator(MiniMigrator):
+    post_migration = True
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        build_reqs = attrs.get("meta_yaml", {}).get("requirements").get("build")
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/cross_compile.py:26]:
```suggestion
        build_reqs = attrs.get("requirements"{}).get("build", set())
```
We expose the requirements top level for ease of use.

File: conda_forge_tick/migrators/cross_compile.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ class UpdateConfigSubGuessMigrator(MiniMigrator):
     post_migration = True
 
     def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
-        build_reqs = attrs.get("meta_yaml", {}).get("requirements").get("build")
+        build_reqs = attrs.get("requirements"{}).get("build", set())
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/cross_compile.py:26]:
```suggestion
        build_reqs = attrs.get("requirements", {}).get("build", set())
```
Sorry! I missed a comma

File: tests/test_cross_compile.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  number: 2
+  run_exports:
+    # change soname at major ver: https://abi-laboratory.pro/tracker/timeline/readline/
+    - {{ pin_subpackage('readline') }}
+
+requirements:
+  build:
+    - pkg-config
+    - {{ compiler('c') }}
+    - make
```
Comment (beckermr) [inline tests/test_cross_compile.py:33]: we should test libtool gets inserted here too, right?

**Comment by isuruf** [comment]
I only added it to the OSX arm migrator so that I can monitor it for a while.

**Pull request merged**